Sources/Network/NetworkAuthenticator.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Authenticator.swift
+//  Alicerce
+//
+//  Created by Luís Portela on 03/07/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public protocol NetworkAuthenticator {
+    func authenticate(request: URLRequest, _ completion: @escaping (URLRequest) -> Void)
```

Comment:
1. Is `completion` the right name for this closure? Isn't it the request performing logic?
2. Since this essentially performs N requests, it should return a `Cancelable`, right?